android/src/main/java/com/aerobotics/DjiMobile/CameraEventDelegate.java

```diff
@@ -33,76 +20,31 @@ public void sendEvent(Object payload) {
   }
 
   private CameraEventObservable cameraEventObservable = new CameraEventObservable();
-
-  private boolean isCameraConnected = false;
   private boolean callbacksSet = false;
-  private final CameraEventDelegate cameraDelegateSenderInstance = this;
-
-  private KeyListener CameraConnectedKeyListener = new KeyListener() {
-    @Override
-    public void onValueChange(@Nullable Object oldValue, @Nullable Object newValue) {
-      if (newValue != null && newValue instanceof Boolean) {
-        if ((boolean) newValue && !isCameraConnected) {
-          isCameraConnected = (boolean)newValue;
-          // When the SDK has just registered, calling getProduct may return null, so wait a second before trying
-          Handler handler = new Handler();
-          handler.postDelayed(new Runnable() {
-            @Override
-            public void run() {
-              cameraDelegateSenderInstance.setCameraCallbacks();
-            }
-          }, 1000);
-        }
-      }
-    }
-  };
 
-  CameraEventDelegate() {
-    boolean cameraConnected = isCameraConnected();
-    if (!cameraConnected) {
-      startCameraConnectionListener();
-    } else {
-      isCameraConnected = true;
-      setCameraCallbacks();
-    }
+  CameraEventDelegate(Camera camera) {
+    setCameraCallbacks(camera);
```

Comment:
Nice clean code style 👏

Reply:
This is infinitely cleaner than what's on the left. Nice one - great "brain wave" 👍